src/GitTools.Core/Helpers/ProcessHelper.cs

```diff
@@ -0,0 +1,144 @@
+﻿namespace GitTools
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+
+    public static class ProcessHelper
+    {
+        private static volatile object _lockObject = new object();
+
+        // http://social.msdn.microsoft.com/Forums/en/netfxbcl/thread/f6069441-4ab1-4299-ad6a-b8bb9ed36be3
+        public static Process Start(ProcessStartInfo startInfo)
+        {
+            Process process;
+
+            lock (_lockObject)
+            {
+                using (new ChangeErrorMode(ErrorModes.FailCriticalErrors | ErrorModes.NoGpFaultErrorBox))
+                {
+                    process = Process.Start(startInfo);
+                    process.PriorityClass = ProcessPriorityClass.Idle;
+                }
+            }
+
+            return process;
+        }
+
+        // http://csharptest.net/532/using-processstart-to-capture-console-output/
+        public static int Run(Action<string> output, Action<string> errorOutput, TextReader input, string exe, string args, string workingDirectory, params KeyValuePair<string, string>[] environmentalVariables)
```

Comment:
I think we should be using threads to read this. I should add an issue to fix this